Fix type definition of Node.js dns.lookup()

shuhei opened the pull request:
- The `options` argument is optional. When it's not specified, the `callback` is the second argument.
- The `callback` is not optional. When it's not specified at all, the function throws a `TypeError`.
- Add specific fields to the `options`.
- Add `dns.ADDRCONFIG` and `dns.V4MAPPED`, which are useful for `hints` in the `options`.

https://nodejs.org/dist/latest-v11.x/docs/api/dns.html#dns_dns_lookup_hostname_options_callback

nmote commented:
This looks reasonable, but unfortunately overloading doesn't always work as expected in Flow. Could you add a few tests to `tests/node_tests` to verify that this has the expected behavior?

shuhei commented:
Sure, I'll add tests for the change and update the PR.

shuhei commented:
I added tests and rebased the branch against `master`. I am not sure about a diff file from `make test`, but I added it as I saw in other pull requests like https://github.com/facebook/flow/pull/7387. Is it what is supposed to do?

shuhei commented:
Thank you for taking care of it!
